selfservice/flow/recovery/flow.go

```diff
@@ -123,13 +127,22 @@ func NewFlow(conf *config.Config, exp time.Duration, csrf string, r *http.Reques
 		Type:      ft,
 	}
 
-	for _, strategy := range strategies {
-		if err := strategy.PopulateRecoveryMethod(r, req); err != nil {
+	// Currently, there are two recovery methods: "code" and (legacy) "link"
```

Comment:
See my comment above in the configuration file. This would avoid such a workaround :)

Reply:
Done.